pkg/kubelet/kubelet.go

```diff
@@ -487,7 +486,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		cadvisor:                                kubeDeps.CAdvisorInterface,
 		cloud:                                   kubeDeps.Cloud,
 		externalCloudProvider:                   cloudprovider.IsExternal(cloudProvider),
-		providerID:                              providerID,
+		providerID:                              kubeCfg.ProviderID,
```

Comment:
`providerID` is always unique per node. Typically you would generate the providerID dynamically by fetching it from a metadata server and then pass it into `--provider-id` flag. You would only do this if you wanted to override the providerID that would be set by the cloud provider. 

I'm not sure how valuable it is inherit this value from kubelet config cause kubelet config is global to all kubelets. If anything, we should maybe deprecate `providerID` in kubelet configuration.

Reply:
Hum 🤔 there is still one kubelet config file per node? Usually they are templated, but yeah I see your point.